sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFn.java

```diff
@@ -462,6 +449,18 @@ public ProcessContinuation processElement(
         // When there are no records available for the current TopicPartition, self-checkpoint
         // and move to process the next element.
         if (rawRecords.isEmpty()) {
+          // Check if the current TopicPartition still exists.
```

Comment:
nit: factor this out to a method?  Then in the future we could more easily add some sort of memoization to that method if desired.

Reply:
Done.